packages/core/src/scripts/migrate-legacy-project.ts

```diff
@@ -0,0 +1,243 @@
+import path from 'path';
+import { promises as fs } from 'fs';
+import type { ManifestData, ImplDeployment } from '../manifest';
+import type { StorageItem, StorageLayout, TypeItem } from '../storage';
+
+const OPEN_ZEPPELIN_FOLDER = '.openzeppelin';
+const MIGRATION_FILE_LOCATION = 'openzeppelin-cli-migration.json';
+
+export async function migrateLegacyProject(): Promise<void> {
```

Comment:
This is missing the deletion of those files that are no longer needed, for example all of the `dev-*` that are renamed to `unknown-*`, although we might want to delete them entirely as I mentioned in another comment.

And there's also the `project.json` which we might want to delete or do something about. This idea goes beyond the requirements that we had initially defined so maybe we want to explore it in a second iteration, but we might want to export the compiler version and options somehow, maybe offering the user to fill in or create their Truffle or Buidler config.

Reply:
Yes that part is yet to be done. Remember this is a draft.